homeassistant/components/iskra/sensor.py

```diff
@@ -163,6 +167,46 @@ class IskraSensorEntityDescription(SensorEntityDescription):
 )
 
 
+def get_counter_sensor(
+    coordinator: IskraDataUpdateCoordinator,
+    counter: Counter,
+    index: int,
+    entity_name: str,
+) -> IskraSensor:
+    """Dynamically create IskraSensor object as energy meter's counters are customizable."""
+
+    key = entity_name.format(index + 1)
+    sensor_description = {
+        "key": key,
+        "translation_key": key,
+        "state_class": SensorStateClass.TOTAL_INCREASING,
+        "native_unit_of_measurement": counter.units,
+    }
+
+    if entity_name == ATTR_NON_RESETTABLE_COUNTER:
+        sensor_description.update(
+            {"value_func": lambda device: device.counters.non_resettable[index].value}
+        )
+    else:
+        sensor_description.update(
+            {"value_func": lambda device: device.counters.resettable[index].value}
+        )
```

Comment:
I think this part should be like
```py
if ...:
    entity_description = IskraSensorEntityDescription()
else:
    entity_description = IskraSensorEntityDescription()
```

and the thing down below can use `.replace` to add the UoM and device class

WDYM?

Reply:
will do

Reply:
with `.replace` you meant like that? 
`entity_description.native_unit_of_measurement = UnitOfEnergy.WATT_HOUR`

Reply:
`dataclasses.replace`